twake/frontend/src/app/services/Apps/Messages/MessageHistoryService.ts

```diff
@@ -0,0 +1,26 @@
+import { Message } from './Message';
+import Numbers from '../../utils/Numbers';
+import FeatureTogglesService, { FeatureNames } from 'app/services/FeatureTogglesService';
+
+class MessageHistoryService {
+  shouldLimitMessages(firstMessageId: string): boolean {
+    const MONTH = 1000 * 60 * 60 * 24 * 30;
+    const DELAY = 3 * MONTH;
+
+    const firstMessageTimestamp = Numbers.timeuuidToDate(firstMessageId) * 1000;
+    const isFirstMessageOlderOrEqualThanDelay = firstMessageTimestamp >= DELAY;
+    const isActiveFeature = FeatureTogglesService.isActiveFeatureName(FeatureNames.MESSAGE_HISTORY);
+    return isFirstMessageOlderOrEqualThanDelay && !isActiveFeature;
+  }
```

Comment:
shouldLimitMessages should return true only when at least N (for instance 100) messages are loaded in the channel. You must add one more parameter to the function.

Reply:
@stephanevieira75

Reply:
Done